### PR TITLE
[quickfix] Add quickfixes for Coq -> Stdlib deprecation warnings.

### PR DESCRIPTION
**Pull request by ejgallego**
This PR should cover all the cases, AFAICT.

I have tested this patch with `coq-lsp`, seems to work fine.

Using the `fcc` compiler, it is possible to apply these fixes automatically to a file.

[Adaptation to QF API revealed the pitfalls of our API entry points
 w.r.t. feedback, errors, etc...]

### CODE REVIEW

**Comment by ejgallego** [comment]
@coqbot run full CI

File: library/nametab.ml [hunk 1]
```diff
@@ -286,14 +286,27 @@ let rec search tree = function
...
   | [] -> false, Some tree.path
 
-let search id tree dir =
+let mk_warn_deprecated_dirpath_Coq ?loc (dir, id) =
+  let quickfix =
+    Option.bind loc (fun loc ->
+        match List.rev dir with
+        | [] -> None
+        | _ :: l ->
+          let dp = DirPath.make (List.rev (Id.of_string "Stdlib" :: l)) in
```
[the remark this answers is not in the document]
Comment (ejgallego) [reply library/nametab.ml:295]: Indeed, I had another look and it'd be better to factorize the code, it is duplicated 3 times.

Will need a bit of tweaking to introduce a wrapper as the CWarnings API doesn't let us to override quickfix directy.

**Comment by herbelin** [comment]
> Using the `fcc` compiler, it is possible to apply these fixes automatically to a file.

Interesting! You mean that `fcc` automatically updates the `.v` file at the same time it produces the `.vo` file?

**Comment by ejgallego** [comment]
> Interesting! You mean that `fcc` automatically updates the `.v` file at the same time it produces the `.vo` file?

Yes, I wrote a small plugin for that, writing such a plugin is very easy:
- hook on document completed event
- grab the list of all diagnostics
- proceed to patch in order.

It is a simple patching so it will likely be not correct in the case of overlapping quickfixes, etc... It also fails if there a choice of qf to make.

Will push to `fcc` repos later on.

**Comment by ejgallego** [comment]
@coqbot run full CI

File: library/nametab.ml [hunk 2]
```diff
@@ -108,13 +108,23 @@ let coq_id = Id.of_string "Coq"
...
-      Pp.(DirPath.(print (make l)) ++ str "." ++ Id.print id ++ spc ()
-          ++ str "has been replaced by" ++ spc ()
-          ++ DirPath.(print (make l')) ++ str "." ++ Id.print id ++ str "."))
+    (fun (old_id, new_id) ->
+      Pp.(old_id ++ spc () ++ str "has been replaced by" ++ spc () ++ new_id ++ str "."))
+
+(* We shadow as to create the quickfix and message at the same time *)
+let fix_coq_id l =
+  (match l with
+   | coq_id :: l -> stdlib_id :: l
```
Comment (proux01) [inline library/nametab.ml:119]: Isn't that a catchall (except for the empty list)?

File: library/nametab.ml (hunk 1 above)
[the remark this answers is not in the document]
Comment (proux01) [reply library/nametab.ml]: Sure, we shouldn't waste too much time on this kind of temporary code.

File: library/nametab.ml (hunk 2 above)
Comment (proux01) [reply library/nametab.ml:119]: Well, I guess it's fine in fact, maybe just
```suggestion
   | _coq_id :: l -> stdlib_id :: l
```
to make it obvious

Comment (ejgallego) [reply library/nametab.ml]: Done thanks

File: test-suite/output/Qf_stdlib.out [hunk 3]
```diff
@@ -0,0 +1,28 @@
+File "./output/Qf_stdlib.v", line 8, characters 15-30:
+Warning: ssr.Coq.ssrbool has been replaced by Stdlib.Coq.ssrbool.
```
Comment (proux01) [inline test-suite/output/Qf_stdlib.out:2]: Is that really the expected result? Looks like some rev went wrong somewhere.

Comment (ejgallego) [reply test-suite/output/Qf_stdlib.out:2]: Yes, the bug was introduced in the refactoring, I had checked this manually but I guess it is easy to not to spot it.

**Comment by ejgallego** [comment]
@coqbot run full CI

**Comment by proux01** [comment]
@coqbot merge now

**Review by proux01 — APPROVED** [review]

**Comment by proux01** [comment]
@coqbot merge now

**Pull request merged**